check_limits.py

```diff
@@ -1,18 +1,96 @@
+def battery_is_ok_EN(temperature, soc, charge_rate):     
+  status_list = [temperature_in_range_EN(temperature), soc_in_range_EN(soc), charge_rate_in_range_EN(charge_rate)]
+  for statuses in status_list:
+    if statuses == False:
+      return False
+  return True
+
+
+def battery_is_ok_DE(temperature, soc, charge_rate):         
+  status_list = [temperature_in_range_DE(temperature), soc_in_range_DE(soc), charge_rate_in_range_DE(charge_rate)]
+  for statuses in status_list:
+    if statuses == False:
+      return False
+  return True
+
+
+def temperature_in_range_EN(temperature):
+  check_warning_EN(0, 45, temperature)
+  if temperature < 0 or temperature > 45:
+    print_statements('Temperature is out of range!')
+    return False
+  return True
 
-def battery_is_ok(temperature, soc, charge_rate):
+
+def temperature_in_range_DE(temperature):
+  check_warning_DE(0, 45, temperature)
   if temperature < 0 or temperature > 45:
-    print('Temperature is out of range!')
+    print_statements('Die Temperatur liegt außerhalb des zulässigen Bereichs!')
     return False
-  elif soc < 20 or soc > 80:
-    print('State of Charge is out of range!')
+  return True
+
+def soc_in_range_EN(soc):
+  check_warning_EN(20, 80, soc)
+  if soc < 20 or soc > 80:
+    print_statements('State of Charge is out of range!')
+    return False
+  return True
+
+
+def soc_in_range_DE(soc):
+  check_warning_DE(20, 80, soc)
+  if soc < 20 or soc > 80:
+    print_statements('Ladezustand außerhalb des Bereichs!')
     return False
-  elif charge_rate > 0.8:
-    print('Charge rate is out of range!')
+  return True
+
+
+def charge_rate_in_range_EN(charge_rate):
```

Comment:
one method per language is not extensible. Right now you have only 2 languages. Imagine the changes required when we add another 10 language support !!

Please think over some other way to handle language strings. maybe you can have a separate strings definition file for each language or an array of strings for each language.